specs/opcode/33CALLER.md

```diff
@@ -0,0 +1,40 @@
+# CALLER opcode
+
+## Procedure
+
+The `CALLER` opcode gets the caller address (msg.caller) from the current call.
+
+## EVM behaviour
+
+The `CALLER` opcode loads an `address` (20 bytes of data) from call context,
+then pushes the `address` to the stack.
+
+## Circuit behaviour
+
+1. Construct call context table in rw table
+2. Do busmapping lookup for stack write operation
+3. Do busmapping lookup for call context caller read operation
+4. Other implicit check: bytes length
+
+## Constraints
+
+1. opId = 0x33
+2. State transition:
+   - gc + 2 (1 stack write, 1 call context read)
+   - stack_pointer - 1
+   - pc + 1
+   - gas + 2
+3. Lookups: 2
+   - `address` is on top of stack
+   - `address` is in the rw table {call context, call ID, caller}
+4. Others:
+   - `address` is 20 bytes length
```

Comment:
I would mention as @han0110 said that this is already constrained by the BeginTx gadget before.

So we're covered.